lib/salus/scanners/report_go_dep.rb

```diff
@@ -1,44 +1,105 @@
 require 'toml'
 require 'salus/scanners/base'
+require 'salus/plugin_manager'
+require 'salus/report'
 
 # Report the use of Go packages captured in a Gopkg.lock files.
 # https://github.com/golang/dep
 
 module Salus::Scanners
   class ReportGoDep < Base
     def run
-      unless should_run?
+      if @repository.go_sum_present?
+        record_dep_from_go_sum
+      elsif @repository.dep_lock_present?
+        record_dep_from_go_lock_package
+      elsif @repository.go_mod_present?
+        record_dep_from_go_mod
+      else
         raise(
           InvalidScannerInvocationError,
-          'Cannot report on Go dependencies without a Gopkg.lock file.'
+          'Cannot report on Go dependencies without a Gopkg.lock, go.mod, or go.sum file'
         )
       end
+    end
+
+    def record_dep_from_go_sum
+      go_sum_path = "#{@repository.path_to_repo}/go.sum"
+      dep_list = []
+
+      File.foreach(go_sum_path).each("=\n") do |line|
+        line = line.strip
+        next if line.empty?
+
+        line_info = get_line_info(line)
+
+        dep_list.append(
+          {
+            "fullDependency" => line,
+            "name" => line_info[0],
+            "version" => line_info[1]
+          }
+        )
+      end
+      # Note references are hashes meant for packages in Gopkg.lock files
+      dep_list.each do |dependency|
+        record_dep_package(
+          name: dependency['name'],
+          reference: "N/A for go.mod/go.sum dependencies",
+          version_tag: dependency['version'],
+          dependency_file: "go.sum",
+          type: "golang"
+        )
+      end
+    end
+
+    def record_dep_from_go_mod
+      # Typically go.mod files don't contain all of the transitive deps/info
+      # a go.sum file would. Instead of parsing go.mod files, warn that
+      # go.sum is missing and send an event
+      warning_string = "WARNING: No go.sum/Gopkg.lock found. "\
+      "Currently go.mod is unsupported for reporting Golang dependencies."
+      report_warn(:report_go_dep_missing_go_sum, warning_string)
+
+      data = "This repository contains no go.sum or Gopkg.lock file. Currently "\
+      "go.mod files are unsupported for reporting Golang dependencies"
+      Salus::PluginManager.send_event(:report_go_dep_scan, data)
```

Comment:
Might be a nice win to move the plugin hook here to within the report_warn. That'd expand the usability to all report warns.